src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php

```diff
@@ -90,8 +89,7 @@ static function ($deferred) {
         $this->invalidateTags = \Closure::bind(
             static function (AdapterInterface $tagsAdapter, array $tags) {
                 foreach ($tags as $v) {
-                    $v->defaultLifetime = 0;
-                    $v->expiry = null;
+                    $v->expiry = 0;
```

Comment:
Does "0" mean "infinity"? then there is a small glitch if one sets `0` as the timestamp for expiry, which will be confused with "infinity".
Should/can we use "-1" for this instead?

Reply:
Yes, according to \Symfony\Component\Cache\Traits\AbstractTrait::doSave() "0" means "infinity". Also, $defaultLifetime = 0 in adapter's constructors mean the same. 

From a user point of view there are only ways to specify expiration:
- $defaultLifetime in adapter. And there 0 is already mean "infinity"
- CacheItemInterface::expiresAfter(). There 0 means number of seconds, so I guess it shouldn't confuse user.
- CacheItemInterface::expiresAt(). Its impossible to pass 0 here.

BTW, this is only place where $item->expiry is set to "0" explicitly, because we want to ignore adapter's default lifetime here if any is set when store tags.

Reply:
> CacheItemInterface::expiresAt(). Its impossible to pass 0 here.

got it: the unix epoch time here is `0.0`, not `0`.